internal/clusterconfig/clusterconfig.go

```diff
@@ -178,6 +210,24 @@ func (r *UpgradeClusterConfigGather) writeProxyToFile(proxy *v1.Proxy, filePath
 	return utils.WriteToFile(p, filePath)
 }
 
+func (r *UpgradeClusterConfigGather) writeMachineConfigsToFiles(mcs []*mcv1.MachineConfig, manifestsPath string) error {
+	for _, machineConfig := range mcs {
+		filePath := filepath.Join(manifestsPath, machineConfig.Name)
+		r.Log.Info("Writing machineConfig to file", "path", filePath)
+		mc := &mcv1.MachineConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   machineConfig.Name,
+				Labels: machineConfig.Labels,
+			},
+			Spec: machineConfig.Spec,
+		}
```

Comment:
If I would `WriteToFIle` directly, the original object's metadata (e.g. `resourceVersion`, `uid`, etc) will be written to the `.json` file as well. The latter would lead to an error when trying to `apply` the manifest later on.

Reply:
Ack, please wrap this logic into a function that has a name that indicates that this is our purpose

Reply:
Usually it makes sense to not override the input/original object (for example to remove some of the ObjectMeta), but since we're anyway using oc apply -f to setup these manifests post-pivot (and not a go client), maybe I could just define a func that removes resourceVersion, uid, creationTimestamp (and maybe generation too) from the original object. WDYT?

Reply:
didn't mean to imply you should change how you do things, just have it in a function so its purpose is clear

Reply:
Looks good